### PR TITLE
feat: Basic captions editing including profanity censoring

### PR DESCRIPTION
**Pull request by philcluff**
## Summary

Adds a new `editCaptions` workflow that edits caption tracks on Mux assets. It supports two independent operations that can be used alone or combined:

- **`autoCensorProfanity`** — LLM-powered profanity detection and censorship. An LLM identifies profane words in the transcript, then word-boundary regex replaces them using one of three modes: `blank` (`[____]`), `remove`, or `mask` (`????`). Supports `alwaysCensor`/`neverCensor` override lists.
- **`replacements`** — static find/replace pairs using case-sensitive word-boundary matching. No LLM needed.

When both are used, profanity censorship runs first (LLM analyses the original text), then static replacements are applied to the post-censorship VTT.

`provider` and `model` are only required when `autoCensorProfanity` is enabled, so pure find/replace usage needs no AI credentials.

By default the edited VTT is uploaded to S3-compatible storage and added as a new subtitle track on the Mux asset, with the original track deleted. Both behaviours are configurable.

Both operations return per-replacement records with cue timing (`ReplacementRecord[]` with `cueStartTime`, `before`, `after`), so callers know exactly what was changed and where.

### Example

```typescript
import { editCaptions } from "@mux/ai/workflows";

const result = await editCaptions("asset-id", "track-id", {
  provider: "anthropic",
  autoCensorProfanity: {
    mode: "blank",
    alwaysCensor: ["brandname"],
    neverCensor: ["damn"],
  },
  replacements: [
    { find: "Mucks", replace: "Mux" },
    { find: "gonna", replace: "going to" },
  ],
  trackNameSuffix: "clean",  // creates "Subtitles (clean)"
});

console.log(result.totalReplacementCount);
console.log(result.autoCensorProfanity?.replacements);
// [{cueStartTime: 5, before: "shit", after: "[____]"}, ...]
console.log(result.replacements?.replacements);
// [{cueStartTime: 30, before: "Mucks", after: "Mux"}, ...]
```

## Suggested review order

1. `src/workflows/edit-captions.ts` — types, pure functions, and the main workflow
2. `tests/unit/edit-captions.test.ts` — unit tests for pure functions and validation
3. `examples/edit-captions/basic-example.ts` — CLI example
4. `docs/API.md` — API reference for the new workflow
5. `docs/WORKFLOWS.md` — workflow guide section
6. Remaining docs (`README.md`, `docs/EXAMPLES.md`, `docs/AUDIO-ONLY.md`) and plumbing (`src/workflows/index.ts`, `package.json`)

---

> [!NOTE]
> **Medium Risk**
> Adds a new workflow that fetches/edits caption tracks and can upload new tracks and delete the original, which is operationally impactful and touches Mux/S3 integration paths. Risk is mitigated by validation and unit tests, but incorrect configuration or edge cases could lead to track loss or unexpected text changes.
> 
> **Overview**
> Introduces a new `editCaptions` workflow to edit existing VTT caption tracks via **LLM-driven profanity detection/censoring** (blank/remove/mask with `alwaysCensor`/`neverCensor` overrides) and/or **static word-boundary find/replace**, returning per-replacement records with cue start times.
> 
> When `uploadToMux` is enabled (default), the workflow uploads the edited VTT to S3-compatible storage, creates a new Mux text track (with configurable name suffix), and optionally deletes the original track; it also supports a non-upload mode that returns the edited VTT only.
> 
> Refactors shared Mux caption helpers into new `lib/mux-tracks.ts` and updates `translateCaptions` to use them, adds unit tests for the editing primitives/validation, and wires in docs + a CLI example plus a new `example:edit-captions` script.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 81195a25684e88d457a103843a1f5b2053d629ed. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: examples/edit-captions/package.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "description": "Examples for @mux/ai caption editing functionality",
+  "scripts": {
+    "basic": "ts-node basic-example.ts"
+  },
+  "dependencies": {
+    "@mux/ai": "file:../..",
+    "dotenv": "^16.0.0"
+  },
```
[the remark this answers is not in the document]
Comment (philcluff) [reply examples/edit-captions/package.json:12]: False positive, it's a dev dependency in the root package.json

File: src/workflows/edit-captions.ts [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+});
+
+/** Inferred shape returned by `profanityDetectionSchema`. */
+export type ProfanityDetectionPayload = z.infer<typeof profanityDetectionSchema>;
+
+// ─────────────────────────────────────────────────────────────────────────────
+// Prompts
+// ─────────────────────────────────────────────────────────────────────────────
+
+const SYSTEM_PROMPT =
```
Comment (walkerfrankenberg) [inline src/workflows/edit-captions.ts]: nit: should we use `dedent` for consistency

File: src/workflows/edit-captions.ts [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+ * Builds a case-insensitive word-boundary regex from an array of profane words.
+ * Words are sorted longest-first so multi-word phrases match before individual words.
+ */
+export function buildReplacementRegex(words: string[]): RegExp | null {
+  const filtered = words.filter(w => w.length > 0);
+  if (filtered.length === 0)
+    return null;
+
+  // Sort by length descending so longer phrases match first
+  const sorted = [...filtered].sort((a, b) => b.length - a.length);
```
Comment (walkerfrankenberg) [inline src/workflows/edit-captions.ts]: any reason we're creating a new array rather than just updating the existing one?

File: src/workflows/edit-captions.ts [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+async function fetchVttFromMux(vttUrl: string): Promise<string> {
+  "use step";
+
+  const vttResponse = await fetch(vttUrl);
+  if (!vttResponse.ok) {
+    throw new Error(`Failed to fetch VTT file: ${vttResponse.statusText}`);
+  }
+
+  return vttResponse.text();
+}
```
[the remark this answers is not in the document]
Comment (walkerfrankenberg) [reply src/workflows/edit-captions.ts]: seems reasonable

File: src/workflows/edit-captions.ts [hunk 5]
```diff
@@ -0,0 +1,649 @@
...
+  }
+
+  if (autoCensorOption && !provider) {
+    throw new Error("provider is required when using autoCensorProfanity.");
+  }
+
+  const deleteOriginal = deleteOriginalOption !== false;
+  const uploadToMux = uploadToMuxOption !== false;
+  const credentials = providedCredentials;
+  const effectiveStorageAdapter = storageAdapter;
```
Comment (walkerfrankenberg) [inline src/workflows/edit-captions.ts]: These seem unnecessary. Can we just name them correctly above?

File: src/workflows/edit-captions.ts [hunk 6]
```diff
@@ -0,0 +1,649 @@
...
+  const s3AccessKeyId = env.S3_ACCESS_KEY_ID;
+  const s3SecretAccessKey = env.S3_SECRET_ACCESS_KEY;
+
+  if (uploadToMux && (!s3Endpoint || !s3Bucket || (!effectiveStorageAdapter && (!s3AccessKeyId || !s3SecretAccessKey)))) {
+    throw new Error(
+      "Storage configuration is required for uploading to Mux. Provide s3Endpoint and s3Bucket. " +
+      "If no storageAdapter is supplied, also provide s3AccessKeyId and s3SecretAccessKey in options " +
+      "or set S3_ENDPOINT, S3_BUCKET, S3_ACCESS_KEY_ID, and S3_SECRET_ACCESS_KEY environment variables.",
+    );
+  }
```
Comment (walkerfrankenberg) [inline src/workflows/edit-captions.ts:441]: not a blocker, but at some point this seems like something we might want to move into a helper

File: src/workflows/edit-captions.ts [hunk 7]
```diff
@@ -0,0 +1,649 @@
...
+  let editedVtt = vttContent;
+  let totalReplacementCount = 0;
+
+  // 1. LLM-powered profanity censorship first (analyses original text)
+  let autoCensorResult: { replacements: ReplacementRecord[] } | undefined;
+  let usage: TokenUsage | undefined;
+  if (autoCensorOption) {
+    const mode = autoCensorOption.mode ?? "blank";
+    const alwaysCensor = autoCensorOption.alwaysCensor ?? [];
+    const neverCensor = autoCensorOption.neverCensor ?? [];
```
Comment (walkerfrankenberg) [inline src/workflows/edit-captions.ts]:
```suggestion
    const { mode = "blank", alwaysCensor = [], neverCensor = [] } = autoCensorOption;
```

File: tests/unit/edit-captions.test.ts [hunk 8]
```diff
@@ -0,0 +1,541 @@
...
+  it("matches exact words with word boundaries", () => {
+    const regex = buildReplacementRegex(["damn"])!;
+    expect("damn".match(regex)).toBeTruthy();
+    expect("damn!".match(regex)).toBeTruthy();
+    expect("oh damn.".match(regex)).toBeTruthy();
+  });
+
+  it("does not match substrings of other words", () => {
+    const regex = buildReplacementRegex(["damn"])!;
+    expect("damage".match(regex)).toBeNull();
```
Comment (walkerfrankenberg) [inline tests/unit/edit-captions.test.ts:134]: nit since the below checks it but this isn't a substring haha

**Review by walkerfrankenberg — APPROVED** [review]
The auto-censor would be able to catch more subtle cases - specifically, multi-word censoring opportunities that are context-dependent, correct?

Otherwise, few comments but no blockers, lgtm

File: src/workflows/edit-captions.ts (hunk 6 above)
Comment (philcluff) [reply src/workflows/edit-captions.ts:441]: Agreed, will put this in jira

File: src/workflows/edit-captions.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (philcluff) [reply src/workflows/edit-captions.ts]: Yep, fixing.
